Add Texture2D.SetDataPointerEXT like on the VertexBuffer

bartwe opened the pull request.

flibitijibibo commented on src/Graphics/Texture2D.cs:
```diff
@@ -158,7 +158,7 @@ int elementCount
 				h = Math.Max(Height >> level, 1);
 			}
-
+			var elementSize = Marshal.SizeOf(typeof(T));
```
Only one `var`, too! Aside from this the patch is good, so just rinse and repeat for Get and 3D/Cube.

bartwe commented:
Added the additional changes and removed the 'var'

Could not change IGLDevice due to the implementations requiring these fields for the other functions which also leaves the SetDataPointerEXT  more verbose in those cases

flibitijibibo commented:
For GetData it looks like you're right; even for buffer objects I opted to stick with the element stuff... SetData, however, I'm not seeing what'd make it more verbose. The GL impls wouldn't use `dataLength`, for example, seeing that they don't use `elementCount`, but VulkanDevice will need it for the memcpy operation and the pointer offsets that are calculated in the Devices can be done by the caller.

So I guess we can leave GetData alone (so that extension can be thrown out, sorry about that!) but SetDataPointer can definitely be applied to all three types.

bartwe commented:
Ok so what do you want me to do with the PR ? remove the GetData ones  and
further nothing ?

On Wed, Jun 27, 2018 at 6:29 PM, Ethan Lee <notifications@github.com> wrote:

> For GetData it looks like you're right; even for buffer objects I opted to
> stick with the element stuff... SetData, however, I'm not seeing what'd
> make it more verbose. The GL impls wouldn't use dataLength, for example,
> seeing that they don't use elementCount, but VulkanDevice will need it
> for the memcpy operation and the pointer offsets that are calculated in the
> Devices can be done by the caller.
>
> So I guess we can leave GetData alone (so that extension can be thrown
> out, sorry about that!) but SetDataPointer can definitely be applied to all
> three types.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FNA-XNA/FNA/pull/182#issuecomment-400742455>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAHJ3qpyFpBirVT6YZ8LWtPgnM3I2lYtks5uA7L7gaJpZM4U56Hx>
> .
>

flibitijibibo commented on src/Graphics/Texture3D.cs:
```diff
@@ -129,14 +130,42 @@ int elementCount
+			IntPtr data,
+			int dataLength
+		) {
+			if (data == null) {
```
Just this one brace, otherwise this is now set to go!

bartwe commented:
My autoformatter hates your autoformatter ;) do you maybe just have a
resharper autoformatter template i can import so it does this automatically
instead of manually ?


On Wed, Jun 27, 2018 at 8:02 PM, Ethan Lee <notifications@github.com> wrote:

> *@flibitijibibo* commented on this pull request.
> ------------------------------
>
> In src/Graphics/Texture3D.cs
> <https://github.com/FNA-XNA/FNA/pull/182#discussion_r198589128>:
>
> >  			);
>  			handle.Free();
>  		}
>
> +		public void SetDataPointerEXT(
> +			int level,
> +			int left,
> +			int top,
> +			int right,
> +			int bottom,
> +			int front,
> +			int back,
> +			IntPtr data,
> +			int dataLength
> +		) {
> +			if (data == null) {
>
> Just this one brace, otherwise this is now set to go!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FNA-XNA/FNA/pull/182#pullrequestreview-132545619>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAHJ3miZINuuNYJpMNLTUcwRJ8XxtAi7ks5uA8jRgaJpZM4U56Hx>
> .
>

flibitijibibo commented:
I don't have any autoformatting locally, just autotabbing, so unfortunately I don't have any templates for anything.

bartwe commented:
pushed whitespace change

flibitijibibo commented:
[Docs have been updated.](https://github.com/FNA-XNA/FNA/wiki/5:-FNA-Extensions#setdatapointerext)
